### DISCUSSION TITLE
Condition for M2M or O2M is null

### DISCUSSION THREAD

**Discussion by Dumpstar56** [directus/directus#13064] [Archive]
Hi there,
I am trying to set a condition for a M2O field called "aggregate." It should be disabled for editing if a M2M relation called "address" is not null. What should I select in the filter menu for this condition?
For conditions for M2O relations on the same table, I select in the filter menu the alias rather than the little arrow to open the related fields to then select isn't null. Those work fine.
If I select the alias on the same table for a M2M relation, it detects isn't null is always true. If I select either of the ID fields in the m2m table, it never detects true. For O2M relationship this also does not work.
What am I missing? Thank you for helping!

**Comment by rijkvanzanten** [comment]
Would you be able to share some more information on the schema setup? I'm having some trouble thinking through the setup based on the provided information 🤔

**Reply by Dumpstar56** [reply]
Sorry for not being clear enough.
In table Camp, I set a condition for field 1. Table Camp also has a M2M relation field to table Camp Type. If I set the relation field as the condition (see attached image) for "is not null," the condition does not detect when the relation is null or not. (This also does not word for O2M relations.)
For an M2O relation, this works. 
![condition m2m issue](https://user-images.githubusercontent.com/94397183/167679687-65782da8-ae2c-4628-b9ed-04ff4b81150f.png)

**Reply by rijkvanzanten** [reply]
Ahh that does make sense (from a technical perspective). The conditions check against the values staged in the app, which means that the many to many is technically never `null` (it will be an empty array rather than null) 🤔

**Comment by domdus** [comment]
Hi @rijkvanzanten, should this work by filtering for _empty then? 
I think I ran into the same issue which seems to be a bug unless my setup is bugged. I need to filter items where related m2m item is empty (m2m item not set).

- create collection `test_a`
- create collection `test_b`
- add m2m to `test_b` that relates to `test_a`
- go to test_b -> create item (+) and just save it (no m2m item set)
- now set a filter in collection test_b, I assume I should filter for "is_empty" ?

<img width="689" alt="image" src="https://user-images.githubusercontent.com/67951828/181506161-1024bb6f-6f85-4ea2-9885-e72a2d6f0aef.png">

Directus 9.14.5

---------------------------------------------------------

Requests failing (500)

/items/test_b?limit=25&fields[]=test_a&fields[]=id&sort[]=id&page=1&filter=%7B%22_and%22:[%7B%22test_a%22:%7B%22_empty%22:true%7D%7D]%7D&meta[]=filter_count&meta[]=total_count

{"errors":[{"message":"select \"test_b\".\"id\" from \"test_b\" where (\"test_b\".\"id\" in (select \"test_b_test_a\".\"test_b_id\" as \"test_b_id\" from \"test_b_test_a\" where \"test_b_test_a\".\"test_b_id\" is not null and (\"test_b_test_a\".\"_empty\" = $1))) order by \"test_b\".\"id\" asc limit $2 - column test_b_test_a._empty does not exist","extensions":{"code":"INTERNAL_SERVER_ERROR"}}]}

/items/test_b?limit=25&filter={"_and":[{"test_a":{"_empty":true}}]}&fields[]=id&fields[]=test_a&sort=id&aggregate={"count":["*"]}

{"errors":[{"message":"select count(*) as \"count\" from \"test_b\" where (\"test_b\".\"id\" in (select \"test_b_test_a\".\"test_b_id\" as \"test_b_id\" from \"test_b_test_a\" where \"test_b_test_a\".\"test_b_id\" is not null and (\"test_b_test_a\".\"_empty\" = $1))) limit $2 - column test_b_test_a._empty does not exist","extensions":{"code":"INTERNAL_SERVER_ERROR"}}]}

**Reply by rijkvanzanten** [reply]
No, `_empty` checks against empty strings. You might be able to utilize the `count` function to check against the number of related items? `count(m2m_field) _eq 0` 🤔

**Reply by domdus** [reply]
Yep, the count function works but it's seems to be a very costly query. (40.000 items in collection) 
It finishes with correct result after like 10+ sec and I'm trying to find a quicker filter setting.

**Comment by Dumpstar56** [comment]
I checked the condition for the o2m alias as disable field if not null again recently, and before saving the item this actually works as intended now. But as soon as the item is saved, no matter whether the relation is set or not, the field is disabled. Not sure what changed, just wanted to post this here to see if that sheds light on anything.
